src/connection.rs

```diff
@@ -221,6 +221,10 @@ pub fn connect(peer_contact: StaticContactInfo,
                     match deserialise::<ListenerResponse>(&read_buf[..bytes_rxd]) {
                         Ok(ListenerResponse::Connect { our_info, their_info, pub_key, }) => {
                             let (their_info, our_info) = (our_info, their_info);
+                            let their_pub_key = pub_key;
```

Comment:
I don't like the idea of renaming the variable on the struct. This struct is serialized and sent over the wire. It has different meanings when it's sent and when it's received.